beprof/curve.py

```diff
@@ -66,19 +66,74 @@ def y_at_x(self, x):
             return x
         return np.interp(x, self.x, self.y, left=np.nan, right=np.nan)
 
+    def change_domain(self, domain):
+
+        # check if new domain includes in the orginal domain
+        if np.max(domain) > np.max(self.x):
+            # maybe some exception here or stderr log ?
+            # these prints only for testing purpose
+            print('Error1')
+            print('domain.max: ', np.max(domain), '|| self.max: ', np.max(self.x))
+            return self
+        if np.min(domain) < np.min(self.x):
+            print('Error2')
+            print('domain.min: ', np.min(domain), '|| self.min: ', np.min(self.x))
+            return self
+        y = np.interp(domain, self.x, self.y)
+        obj = Curve([[domain[ind], y[ind]] for ind in range(0, len(y))])
```

Comment:
This is inefficient and is not python. Why don't you use zip or similar method from numpy. Read http://docs.scipy.org/doc/numpy-1.10.1/reference/routines.array-manipulation.html (section "Joining arrays").